### DISCUSSION TITLE
Checkbox widget with side-by-side checkmarks

### DISCUSSION THREAD

**Discussion by Nikoichu** [iSoron/uhabits#1329] [Feature Requests]
## Pre-submission checklist
- [ ] I am submitting a bug report, not a feature request.
- [X] I have searched for similar issues, but did not find any matches.

## Current Behavior
Currently, when creating a simple checkbox widget, you can select multiple habits to show on it. However, the habits get stacked ontop of each other and its almost impossible to click the ones behind. Scaling the widget doesn't help either - it doesn't spread the checkboxes like I'd expect it to, it just scales them up, while remaining stacked.

## Proposed behavior
Make another widgwt just for this purpose that's sized 2x2 by default, and spreads the checkmarks evenly across the space it's allotted, with a connected background.

This way the checkboxes will look much nicer with a connected background and be much more compact than having to create 6 single checkboxes side-by-side, taking 6 homescreen spaces and having separated backgrounds. (My current setup lol)

## System information
 App version 2.0.3
Android version 10

## P.S.
Okay I just found out you can swipe down to change the habit on top... This should be better documented (or I'm slightly stupid lol). However, I still think a widget that shows all boxes side-by-side would be useful.

**Comment by hiqua** [comment]
What would be the difference with just creating new widgets, one habit per widget?

**Reply by Khedde** [reply]
I've done this, and it was absolute hell creating and rearranging each widget. I'd imagine it could be even worse depending on which launcher you use. 
This frustration scales exponentially the more individual widgets you have to make. 
Terrible experience, would not recommend.

**Reply by hiqua** [reply]
I don't get why your experience was so bad. In my case, I just kept creating individual checkmark widgets until I had enough. Anyway it's not something you're supposed to do regularly, so I'd say the current setup is fine, even if it could be better.

**Reply by Nikoichu** [reply]
Besides the practical reason above, there is also space-efficiency (not every launcher has a lot of space for icons and widgets on each screen) and then there is the aesthetic reason - one connected background would look so much better and cleaner.

**Reply by hiqua** [reply]
> space-efficiency (not every launcher has a lot of space for icons and widgets on each screen)

On my phone, I can go to the theme settings, and select a grid of 6x6 which gives plenty of space for small icons. I agree that it could look better with a unified background though.
